lib/array_equals.rb

```diff
@@ -1,5 +1,18 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  if array1 == nil && array2 == nil || array1 == [] && array2 == []
+    return true
+  elsif array1 == nil || array2 == nil || array1.length != array2.length
+    return false
+  else
+    i = 0
+    array1.each do |x|
```

Comment:
`.each` iterates through **arrays** in order in Ruby, but [may not for other data types](https://stackoverflow.com/a/6590245)

Would this break if (for whatever reason) `.each` did its iteration out of order? If so, how might we fix that?